source/MaterialXGenShader/ShaderGraph.cpp

```diff
@@ -1069,9 +1064,11 @@ void ShaderGraph::optimize(GenContext& context)
         // Traverse the graph to find nodes still in use
         for (ShaderGraphOutputSocket* outputSocket : getOutputSockets())
         {
-            if (outputSocket->getConnection())
+            // Make sure to not include connections to the graph itself.
+            ShaderOutput* upstreamPort = outputSocket->getConnection();
+            if (upstreamPort && upstreamPort->getNode() != this)
```

Comment:
Fixing a bug where we didn't check if the connection upstream was to the graph itself (the graph interface). So the graph itself could potentially be added as a node in the graph here.